client/my-sites/site-settings/traffic/main.jsx

```diff
@@ -51,12 +64,26 @@ SiteSettingsTraffic.propTypes = {
 };
 
 const connectComponent = connect(
-	( state ) => ( {
-		site: getSelectedSite( state ),
-	} )
+	( state ) => {
+		const site = getSelectedSite( state );
+		const siteId = getSelectedSiteId( state );
+		const isJetpack = isJetpackSite( state, siteId );
+		const jetpackSettingsUiSupported = isJetpack && siteSupportsJetpackSettingsUi( state, siteId );
+
+		return {
+			site,
+			jetpackSettingsUiSupported,
+		};
+	}
 );
 
 const getFormSettings = partialRight( pick, [
+	'stats',
+	'admin_bar',
+	'hide_smile',
+	'count_roles',
+	'roles',
+
```

Comment:
At the time it was meant to be there for separating settings that come from different modules.  But now that you mention it, it's probably not necessary anyway, so I've removed it.